repository/firestore.go

```diff
@@ -0,0 +1,97 @@
+package repository
+
+import (
+	"cloud.google.com/go/firestore"
+	"context"
+	"errors"
+	"github.com/gazebo-web/gz-go/v7/reflect"
+)
+
+var (
+	// ErrMethodNotImplemented is returned when a certain method is not implemented.
+	ErrMethodNotImplemented = errors.New("method not implemented")
+)
+
+// firestoreRepository implements Repository using the firestore client.
+type firestoreRepository[T Model] struct {
+	entity Model
+	client *firestore.Client
+}
+
+// FirstOrCreate is not implemented.
+func (f *firestoreRepository[T]) FirstOrCreate(entity Model, filters ...Filter) error {
+	return ErrMethodNotImplemented
+}
+
+// Create is not implemented.
+func (f *firestoreRepository[T]) Create(entity Model) (Model, error) {
+	return nil, ErrMethodNotImplemented
+}
+
+// CreateBulk is not implemented.
+func (f *firestoreRepository[T]) CreateBulk(entities []Model) ([]Model, error) {
+	return nil, ErrMethodNotImplemented
+}
+
+// Find filters entries and stores filtered entries in output.
+// output: will contain the result of the query. It must be a pointer to a slice.
+// options: configuration options for the search.
+func (f *firestoreRepository[T]) Find(output interface{}, options ...Option) error {
```

Comment:
I understand that this is an early draft, but will this PR include the necessary options to paginate results?

Reply:
I was planning on adding a child PR in order to add support for pagination.

Reply:
Sounds like a plan.